test/unit/transaction.test.js

```diff
@@ -23,528 +16,534 @@ const Segment = require('../../lib/transaction/trace/segment')
-        agent.transactionNameNormalizer.addSimple('^(WebTransaction/Custom)/test$', '$1/*')
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 200)
-        expect(trans.name).equal('WebTransaction/Custom/*')
-      })
+    t.test('produces a `method not allowed` name when status is 405', (t) => {
+      txn.nameState.setName('Expressjs', 'GET', '/')
+      txn.finalizeNameFromUri('/test/string?do=thing&another=thing', 405)
+      t.equal(txn.name, 'WebTransaction/Expressjs/GET/(method not allowed)')
+      t.end()
+    })
 
-      it('passes through status code when status is 200', function () {
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 200)
-        expect(trans.statusCode).equal(200)
-      })
+    t.test('produces a name based on 501 status code message', (t) => {
+      txn.finalizeNameFromUri('/test/string?do=thing&another=thing', 501)
+      t.equal(txn.name, 'WebTransaction/WebFrameworkUri/(not implemented)')
+      t.end()
+    })
 
-      it('keeps the custom name when error status is ignored', function () {
-        agent.config.error_collector.ignore_status_codes = [404, 500]
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 500)
-        expect(trans.name).equal('WebTransaction/Custom/test')
-      })
+    t.test('produces a regular partial name based on 501 status code message', (t) => {
+      txn.finalizeNameFromUri('/test/string?do=thing&another=thing', 501)
+      t.equal(txn._partialName, 'WebFrameworkUri/(not implemented)')
+      t.end()
+    })
 
-      it('keeps the custom partial name when error status is ignored', function () {
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 404)
-        expect(trans.nameState.getName()).equal('Custom/test')
-      })
+    t.test('passes through status code when status is 501', (t) => {
+      txn.finalizeNameFromUri('/test/string?do=thing&another=thing', 501)
+      t.equal(txn.statusCode, 501)
+      t.end()
+    })
 
-      it('passes through status code when status is 404', function () {
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 404)
-        expect(trans.statusCode).equal(404)
-      })
+    t.test('should update value from segment normalizer rules', (t) => {
+      const url = 'NormalizedUri/test/explicit/string/lyrics'
+      txn.forceName = url
+      txn.url = url
+      agent.txSegmentNormalizer.load([
+        { prefix: 'WebTransaction/NormalizedUri', terms: ['test', 'string'] }
+      ])
+      txn.finalizeNameFromUri(url, 200)
+      t.equal(txn.name, 'WebTransaction/NormalizedUri/test/*/string/*')
+      t.end()
+    })
 
-      it('produces the custom name even when status is 501', function () {
-        trans.finalizeNameFromUri('/test/string?do=thing&another=thing', 501)
-        expect(trans.name).equal('WebTransaction/Custom/test')
-      })
+    t.test('should not scope web transactions to their URL', (t) => {
```

Comment:
Is this subtest new for this group?

Reply:
Yes, but I think it belongs with this group. Originally it was alone at the "top level". We are checking that its name isn't scoped. In particular, when a partial name isn't set.